### PR TITLE
community: Added integrations for ThirdAI's NeuralDB with Retriever and VectorStore frameworks

### PR DESCRIPTION
**Pull request by benitoThree**
**Description:** Adds ThirdAI NeuralDB retriever and vectorstore integration. NeuralDB is a CPU-friendly and fine-tunable text retrieval engine.

### CODE REVIEW

File: libs/langchain/tests/integration_tests/.env.example [hunk 1]
```diff
@@ -34,4 +34,8 @@ POWERBI_NUMROWS=_num_rows_in_your_test_table
 
 
 # MongoDB Atlas Vector Search
-MONGODB_ATLAS_URI=your_mongodb_atlas_connection_string
\ No newline at end of file
+MONGODB_ATLAS_URI=your_mongodb_atlas_connection_string
+
+# ThirdAI NeuralDB
+# Request a license at https://www.thirdai.com/try-bolt/
+THIRDAI_KEY=your-thirdai-key-here
```
Comment (hwchase17) [inline libs/langchain/tests/integration_tests/.env.example:41]: probably doesnt need a change? changes are all to community right?

Comment (benitoThree) [reply libs/langchain/tests/integration_tests/.env.example:41]: Yes, all changes are in community. Is this file only for tests in core? Are there other options for integration other than community? To my understanding, all third-party integrations go under langchain-community unless we want to have our own separate package. I'm a little confused about how users are going to import the integrations. For example, in the [elastic search description](https://integrations.langchain.com/vectorstores?integration_name=ElasticVectorSearch), it says `from langchain.vectorstores import ElasticVectorSearch` at the top but `from langchain_community.vectorstores import ElasticVectorSearch` in the description. Are integrations from `lanchain_community` automatically accessible from `langchain`?

Comment (hwchase17) [reply libs/langchain/tests/integration_tests/.env.example:41]: they are for now, but the entrypoints in langchain are getting deprecated

**Comment by benitoThree** [comment]
Hi @hwchase17, I just wanted to check in on what the rest of the review process will look like. Is there anything we need to do before review / merging? Thank you!

**Comment by benitoThree** [comment]
Hi @baskaryan, I just wanted to check in on what the rest of the review process will look like. Is there anything we need to do before review / merging? Thank you!

**Review by hwchase17 — COMMENTED** [review]
why do we need both the retriever and the vectorstore classes? all vectorstores can automatically be converted to retrievers with `vectorstore.as_retriever()`... so why do we need boht?

**Comment by benitoThree** [comment]
@hwchase17 My bad! I didn't realize there was an as_retriever() option. Updated it to only have vectorstore

**Review by hwchase17 — APPROVED** [review]

**Pull request merged**